app/features-json/repos_json_controller.rb

```diff
@@ -16,5 +17,57 @@ class RepositoryJSONController < AuthenticatedControllerBase
 
       return all_repos_view_models.to_json
     end
+
+    # Accepts the repo full name and branch as query params
+    # Ex. "../repos/lanes?repo_full_name=nakhbari%2FHelloWorld&branch=master"
+    get "#{HOME}/lanes" do
+      repo_full_name = params[:repo_full_name]
+      branch = params[:branch]
+
+      provider_credential = check_and_get_provider_credential(
+        type: FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github]
+      )
+
+      github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
+
+      selected_repo = github_service.repos.detect do |repo|
+        repo_full_name == repo[:full_name]
+      end
+
+      if selected_repo.nil?
+        raise "Could not find repo, check that your github token has access to repo: #{repo_full_name}}"
```

Comment:
In this case, we're using the user's GitHub token, not the fastlane bot account token, right? If so, maybe we can change the messaging to mention which GH token was tried to be used?